File: Python/Test/jdi_uitests_webtests/test/common/text_area_tests.py
from JDI.jdi_assert.testing.assertion import Assert
from Test.jdi_uitests_webtests.main.enums.preconditions import Preconditions
from Test.jdi_uitests_webtests.main.page_objects.epam_jdi_site import EpamJDISite
from Test.jdi_uitests_webtests.test.init_tests import InitTests


class TextAreaTests(InitTests):
    LINES = ["line1", "line2", "line3"]
    text_item = EpamJDISite.contact_form_page.description

    def setUp(self):
        Preconditions.CONTACT_PAGE.is_in_state()

    def test_input_line(self):
        self.text_item.input_lines(self.LINES)
        lines = self.text_item.get_lines()
        Assert.assert_equal(lines, self.LINES)

    def test_add_new_line(self):
        new_line = "newLine"
        self.text_item.send_keys("garbageString")
        self.text_item.add_new_line(new_line)
        lines = self.text_item.get_lines()
        Assert.assert_equal(new_line, lines[1])

    def test_get_lines(self):
        self.text_item.send_keys(self.LINES[0] + "\n" + self.LINES[1] + "\n" + self.LINES[2])
        Assert.assert_equal(self.text_item.get_lines(), self.LINES)

File: Java/Tests/jdi-uitest-webtests/src/test/java/com/epam/jdi/uitests/testing/unittests/tests/common/TextAreaTests.java
package com.epam.jdi.uitests.testing.unittests.tests.common;

import com.epam.jdi.uitests.testing.unittests.InitTests;
import com.epam.jdi.uitests.web.selenium.elements.common.TextArea;
import org.testng.Assert;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Factory;
import org.testng.annotations.Test;

import java.lang.reflect.Method;

import static com.codeborne.selenide.Condition.*;
import static com.epam.jdi.uitests.core.preconditions.PreconditionsState.isInState;
import static com.epam.jdi.uitests.testing.unittests.entities.User.DEFAULT;
import static com.epam.jdi.uitests.testing.unittests.enums.Preconditions.CONTACT_PAGE;
import static com.epam.jdi.uitests.testing.unittests.enums.Preconditions.CONTACT_PAGE_FILLED;
import static com.epam.jdi.uitests.testing.unittests.pageobjects.EpamJDISite.contactFormPage;
import static com.epam.jdi.uitests.web.settings.WebSettings.getJSExecutor;
import static org.testng.Assert.assertEquals;

/**
 * Created by Dmitry_Lebedev1 on 11/12/2015.
 */
public class TextAreaTests extends InitTests {
    final String[] LINES = {"line1", "line2", "line3"};
    private TextArea textItem() { return contactFormPage.description; }

    private String[] getStringArray() {
        return getJSExecutor().executeScript("return arguments[0].value", textItem().getWebElement()).toString().split("\n");
    }

    @BeforeMethod
    public void before(final Method method) {
        isInState(CONTACT_PAGE, method);
    }

    @Test
    public void inputLinesTest() {
        textItem().inputLines(LINES);
        String[] out = getStringArray();
        textItem().getText();
        assertEquals(LINES, out);
    }

    @Test
    public void addNewLineTest() {
        textItem().sendKeys("garbageString");
        String newLine = "newLine";
        textItem().addNewLine(newLine);
        String[] out = getStringArray();
        assertEquals(newLine, out[1]);
    }

    @Test
    public void getLinesTest() {
        textItem().sendKeys(LINES[0] + "\n" + LINES[1] + "\n" + LINES[2]);
        assertEquals(textItem().getLines(), LINES);
    }

    @Test
    public void getLinesLengthTest() {
        textItem().sendKeys(LINES[0] + "\n" + LINES[1] + "\n" + LINES[2]);
        assertEquals(textItem().getLines().length, 3);
    }

    @Factory
    public Object[] textTests() {
        return new Object[]{
                new TextFieldTests("ITextArea", CONTACT_PAGE_FILLED,
                        this::textItem, "text123", "text123",
                        DEFAULT.description, "pti", ".escriptio.")
        };
    }

    @Test
    public void imageIsDisplayedTest(){
        Assert.assertTrue(textItem().isDisplayed());
    }

    @Test
    public void shouldTest(){
        textItem().shouldHave(attribute("rows"), attribute("cols"), id("Description"))
                .shouldBe(empty);
    }
}
